torch/_inductor/lowering.py

```diff
@@ -2001,10 +2001,71 @@ def apply_constraint(arg, fx_arg):
 make_fallback(aten._fused_moving_avg_obs_fq_helper_functional)
 make_fallback(aten.grid_sampler_2d_backward, require_dense)
 make_fallback(aten.randperm)
-make_fallback(aten._scaled_dot_product_efficient_attention)
-make_fallback(aten._scaled_dot_product_efficient_attention_backward)
-make_fallback(aten._scaled_dot_product_flash_attention, warn=False)
-make_fallback(aten._scaled_dot_product_flash_attention_backward)
+
+
+def sdpa_constraint(fx_node, *args, **kwargs):
+    # sdpa requires dense last dimension
+    def apply_constraint(arg, fx_arg):
+        if not isinstance(arg, ir.IRNode):
+            return arg
+
+        meta_val = fx_arg.meta["val"]
+        if not meta_val.is_cuda:
+            return arg
+
+        stride_order = ir.get_stride_order(meta_val.stride())
+        if stride_order and stride_order[-1] != 0:
+            stride_order = list(reversed(range(len(arg.get_size()))))
+
+        efficient_kernels = (
+            aten._scaled_dot_product_efficient_attention.default,
+            aten._scaled_dot_product_efficient_attention_backward.default,
+        )
+        ALIGNMENT = 16 if fx_node.target in efficient_kernels else 8
```

Comment:
why not only 16?

Reply:
correct me if i'm wrong here. i was getting an error for 8 alignment in cutlass. should it be 16 ?

Reply:
https://github.com/pytorch/pytorch/blob/b70efde3ad6ee7385eab9fa3d40ae45236c9e4a3/aten/src/ATen/native/transformers/attention.cpp#L544

I have been using 16 which I think is the requirement